server/src/main/java/com/flipkart/varadhi/web/v1/authz/AuthZHandlers.java

```diff
@@ -0,0 +1,205 @@
+
+import static com.flipkart.varadhi.Constants.PathParams.REQUEST_PATH_PARAM_RESOURCE;
+import static com.flipkart.varadhi.Constants.PathParams.REQUEST_PATH_PARAM_RESOURCE_TYPE;
+import static com.flipkart.varadhi.web.routes.RouteBehaviour.authenticated;
+import static com.flipkart.varadhi.web.routes.RouteBehaviour.hasBody;
+
+@Slf4j
+@ExtensionMethod({Extensions.RequestBodyExtension.class, Extensions.RoutingContextExtension.class})
+public class AuthZHandlers implements RouteProvider {
+    private final AuthZService authZService;
+
+    public AuthZHandlers(AuthZService authZService) {
+        this.authZService = authZService;
+    }
+
+    private List<RouteDefinition> getDebugHandlers() {
+        return new SubRoutes(
+                "/v1/authz/debug",
+                List.of(
+                        new RouteDefinition(
+                                HttpMethod.GET,
+                                "",
+                                Set.of(authenticated),
+                                new LinkedHashSet<>(),
+                                this::getAllRoleBindingNodes,
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.GET,
+                                "/:resource_type/:resource",
+                                Set.of(authenticated),
+                                new LinkedHashSet<>(),
+                                this::findRoleBindingNode,
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.DELETE,
+                                "/:resource_type/:resource",
+                                Set.of(authenticated),
+                                new LinkedHashSet<>(),
+                                this::deleteRoleBindingNode,
+                                true,
+                                Optional.empty()
+                        )
+                )
+        ).get();
+    }
+
+    public List<RouteDefinition> getPolicyHandlers() {
+        return new SubRoutes(
+                "/v1",
+                List.of(
+                        new RouteDefinition(
+                                HttpMethod.GET,
+                                "/orgs/:org/policy",
+                                Set.of(authenticated),
+                                new LinkedHashSet<>(),
+                                this.getIAMPolicyHandler(ResourceType.ORG),
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.PUT,
+                                "/orgs/:org/policy",
+                                Set.of(hasBody, authenticated),
```

Comment:
Since there are repetitions here,
we can put common patterns in the RouteDefinition itself  as static methods.

```java
public static Builder get(String path) {
        return new Builder(HttpMethod.GET, path);
    }

    public static Builder put(String path) {
        return new Builder(HttpMethod.PUT, path);
    }

    public static Builder post(String path) {
        return new Builder(HttpMethod.POST, path);
    }

    public static Builder delete(String path) {
        return new Builder(HttpMethod.DELETE, path);
    }

    @RequiredArgsConstructor
    public static class Builder {

        private final HttpMethod method;

        private final String path;

        private boolean authenticated;

        private boolean hasBody;

        private boolean blocking;

        public Builder authenticated() {
            this.authenticated = true;
            return this;
        }

        public Builder hasBody() {
            this.hasBody = true;
            return this;
        }

        public Builder blocking() {
            this.blocking = true;
            return this;
        }

        public RouteDefinition build(Handler<RoutingContext> reqHandler) {
            Set<RouteBehaviour> behaviours = new LinkedHashSet<>();
            if (authenticated) {
                behaviours.add(RouteBehaviour.authenticated);
            }
            if (hasBody) {
                behaviours.add(RouteBehaviour.hasBody);
            }
            return new RouteDefinition(
                    method,
                    path,
                    behaviours,
                    new LinkedHashSet<>(),
                    reqHandler,
                    blocking,
                    Optional.empty()
            );
        }
    }
```

Then these usages will look like
```java
RouteDefinition.get("/orgs/:org/policy")
                                .authenticated()
                                .blocking()
                                .build(this.getIAMPolicyHandler(ResourceType.ORG)),
```

Reply:
done